mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp

```diff
@@ -0,0 +1,87 @@
+//===- Pipeline.cpp - Create MIOpen compilation pipeline
+//------------------===//
+//
+// Copyright 2020 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/MIOpen/Pipeline.h"
+
+#include "mlir/Conversion/MIOpenPasses.h"
+#include "mlir/Dialect/MIOpen/Passes.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/InitAllPasses.h"
+
+#include "mlir/Conversion/GPUToROCDL/GPUToROCDLPass.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVM.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVMPass.h"
+#include "mlir/InitAllDialects.h"
+#include "llvm/Support/TargetSelect.h"
+
+using namespace mlir;
+
+//===- miopen::createPipeline -----------------------------------===//
+//===- Consolidate the MIOpen Pipeline here ---------------------===//
+
+void miopen::addPipeline(PassManager &pm, const PipelineConfig &pc) {
+  if (pc.highLevelInput) {
```

Comment:
Prefer to be more specific in parameter naming, `tosaInput` is better than `highLevelInput`

Reply:
I will make an enum since it will soon include MIXR as well.